tests/test_utils.py

```diff
@@ -68,3 +50,34 @@ def test_find_upward_start_path_none(mocker):
 
     # Assert the result is None since the mocked directory does not contain the target
     assert result is None
+
+
+def test_find_upward_returns_correct_path(temp_dir_structure):
+    # Test that the function returns the correct path to the target
+    result = find_upward("target.txt", start_path=temp_dir_structure["sub_dir"])
+    assert result == temp_dir_structure["root_dir"]
+
+
+def test_find_upward_handles_symlinks(temp_dir_structure):
+    # Create a symlink to the target file
+    symlink_path = os.path.join(temp_dir_structure["sub_dir"], "symlink_to_target.txt")
+    os.symlink(temp_dir_structure["target_file"], symlink_path)
+
+    # Test finding the symlink
+    result = find_upward("symlink_to_target.txt", start_path=temp_dir_structure["sub_dir"])
+    assert result == temp_dir_structure["sub_dir"]
+
+    # Clean up the symlink
+    os.remove(symlink_path)
```

Comment:
Just making sure of the intent of this test.  Is this to ensure that `find_upward` doesn't follow a link, but treats it as a valid target?

Reply:
yes.

Reply:
don't want to follow the link, just return to where that link is.